modin/engines/base/io/io.py

```diff
@@ -11,51 +11,111 @@
 # ANY KIND, either express or implied. See the License for the specific language
 # governing permissions and limitations under the License.
 
+"""
+Module houses `BaseIO` class.
+
+`BaseIO` is base class for IO classes, that stores IO functions.
+"""
+
 import pandas
+from pandas.util._decorators import doc
 from collections import OrderedDict
 from modin.error_message import ErrorMessage
 from modin.backends.base.query_compiler import BaseQueryCompiler
+from modin.utils import _inherit_docstrings
 from typing import Optional
 
+_doc_default_io_method = """
+{summary} using pandas.
+
+For parameters description please refer to pandas API.
+
+Returns
+-------
+{returns}
+"""
+
+_doc_returns_qc = """BaseQueryCompiler
+    QueryCompiler with read data."""
+
+_doc_returns_qc_or_parser = """BaseQueryCompiler or TextParser
+    QueryCompiler or TextParser with read data."""
+
 
 class BaseIO(object):
+    """Class for basic utils and default implementation of IO functions."""
+
     query_compiler_cls: BaseQueryCompiler = None
     frame_cls = None
 
     @classmethod
     def from_non_pandas(cls, *args, **kwargs):
+        """
+        Create a Modin `query_compiler` from a non-pandas `object`.
+
+        Parameters
+        ----------
+        *args : iterable
+            Positional arguments to be passed into `func`.
+        **kwargs : dict
+            Keyword arguments to be passed into `func`.
+        """
         return None
 
     @classmethod
     def from_pandas(cls, df):
+        """
+        Create a Modin `query_compiler` from a `pandas.DataFrame`.
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            The pandas DataFrame to convert from.
+
+        Returns
+        -------
+        BaseQueryCompiler
+            QueryCompiler containing data from the `pandas.DataFrame`.
+        """
         return cls.query_compiler_cls.from_pandas(df, cls.frame_cls)
 
     @classmethod
     def from_arrow(cls, at):
+        """
+        Create a Modin `query_compiler` from a `pyarrow.Table`.
+
+        Parameters
+        ----------
+        at : Arrow Table
+            The Arrow Table to convert from.
+
+        Returns
+        -------
+        BaseQueryCompiler
+            QueryCompiler containing data from the Arrow Table.
+        """
         return cls.query_compiler_cls.from_arrow(at, cls.frame_cls)
 
     @classmethod
+    @_inherit_docstrings(pandas.read_parquet, apilink="pandas.read_parquet")
+    @doc(
```

Comment:
Both decorators change `__doc__`, how are they supposed to work together?

Reply:
`_inherit_docstrings` decorator only appends docstring generated by `doc` decorator since we don't explicitly set `overwrite_existing` parameter of `_inherit_docstrings` like it is done [here](https://github.com/modin-project/modin/blob/master/modin/pandas/dataframe.py#L60).

Reply:
Got it.

Reply:
`@_inherit_docstrings` _appends_ to existing `__doc__` unless instructed to overwrite.